lighthouse/tests/validator_client.rs

```diff
@@ -260,6 +260,7 @@ fn http_flag() {
 fn http_address_flag() {
     let addr = "127.0.0.99".parse::<IpAddr>().unwrap();
     CommandLineTest::new()
+        .flag("http", None)
```

Comment:
Are these added to fix some new failing tests?
It makes sense but I'm just asking because the cli definition doesn't actually require "http" to be present when the below flags are used (e.g. "http-address" arg doesn't have `.require("http")`)

https://github.com/sigp/lighthouse/blob/1373dcf076bd031e1c09119281c5f84b789974c2/validator_client/src/cli.rs#L170-L188

Reply:
yes, thanks for bringing this up Jimmy. without the `http` and the `metrics` flag the both servers don't run,
see here for the validator client http server:
https://github.com/sigp/lighthouse/blob/dfcb3363c757671eb19d5f8e519b4b94ac74677a/validator_client/src/lib.rs#L570-L584 
and here for the metrics:
https://github.com/sigp/lighthouse/blob/dfcb3363c757671eb19d5f8e519b4b94ac74677a/validator_client/src/lib.rs#L127-L155
I think I commented on mattermost that we should probably make all these flags dependent on the main `http` and `metrics` ones, I didn't know it was that simple, I can do that in a subsequent PR if everyone agrees.

Reply:
These tests actually just verify the parsed configs. The `CommandLineTest` starts the lighthouse process, dump the loaded config and shuts down immediately (it doesn't even create a `ProductionBeaconNode` or `ProductionValidatorClient`). But if were to make these flags required, we'll need them here. Updating in a subsequent PR is fine by me!

https://github.com/sigp/lighthouse/blob/b6493d5e2400234ce7148e3a400d6663c3f0af89/lighthouse/tests/exec.rs#L37-L43

Reply:
>These tests actually just verify the parsed configs.

Yeah but it was one of the tests that already had the `metrics` flag before that failed and allowed us to acknowledge the CORS bug after the warp update, that's why I added the flag to all the other tests.

>But if were to make these flags required, we'll need them here. Updating in a subsequent PR is fine by me!

Ok nice, I'll do it then, thanks Jimmy